workflows/index-generation/ncbi-compress/src/ncbi_compress.rs

```diff
@@ -346,45 +345,87 @@ pub mod ncbi_compress {
     ) {
         let reader = fasta::Reader::from_file(&input_fasta_path).unwrap();
         // let mut tree: BloomSetTree<_, 4096> = BloomSetTree::new(branch_factor);
-        let mut sketches = Vec::new();
+        let mut sketches: Vec<KmerMinHash> = Vec::new();
 
-        // Initialize a temporary vector to store the unique items from each chunk
-        let mut tmp: Vec<KmerMinHash> = Vec::with_capacity(chunk_size);
+        let writer = Mutex::new(writer);
 
         let mut records_iter = reader.records();
-        let mut chunk = records_iter
-            .borrow_mut()
-            .take(chunk_size)
-            .collect::<Vec<_>>();
-        accession_count.add_assign(chunk.len() as u64);
-        while chunk.len() > 0 {
-            // Take a chunk of accessions and return the ones that are not already in sketches
+
+
+        loop {
+            // Initialize a temporary vector to store the unique items from each chunk
+            let mut unique_in_chunk: Vec<(KmerMinHash, &fasta::Record)> = Vec::with_capacity(chunk_size);
+            let mut unique_in_tree_and_chunk: Vec<KmerMinHash> = Vec::with_capacity(chunk_size);
```

Comment:
Slight performance thing by declaring these outside of the loop than clearing it will save you allocations.

Reply:
fixed